src/main/java/cat/udl/eps/softarch/mytournamentx/domain/TournamentInvitation.java

```diff
@@ -1,20 +1,58 @@
 package cat.udl.eps.softarch.mytournamentx.domain;
 
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
+import org.hibernate.validator.constraints.Length;
+import org.springframework.format.annotation.DateTimeFormat;
 
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.time.ZonedDateTime;
 
 @Entity
 @Data
 @EqualsAndHashCode(callSuper = true)
-public class TournamentInvitation extends UriEntity<Integer> {
+public class TournamentInvitation extends UriEntity<Integer>{
 
     @Id
-    @GeneratedValue(strategy= GenerationType.AUTO)
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
     private Integer id;
 
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    private ZonedDateTime createdAt;
+
+    @ManyToOne
+    @JsonIdentityReference(alwaysAsId = true)
+    private Player invites;
+
+    @ManyToOne
+    @JsonIdentityReference(alwaysAsId = true)
+    private Player createdBy;
+
+    @ManyToOne
+    @JsonIdentityReference(alwaysAsId = true)
+    private Tournament invitesTo;
+
+    @Lob
+    @NotBlank
+    @Size(max = 255)
+    private String message;
+
+    public TournamentInvitation(@NotBlank @Length(min = 1, max = 250) String message) {
```

Comment:
Redundant annotations. Moreover, Spring won't call this constructor so this annotations would never have effect.